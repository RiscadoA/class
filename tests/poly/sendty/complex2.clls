proc main() {
    cut {
        println("sendty"); sendty x(lint);
        println("lhs send"); send x(y. println("let y"); let y 7);
        println("lhs recv"); recv x(y);
        println("lhs wait"); wait x;
        println(y); ()
        |x: recvty T; recv ~T; send T; close|
        println("recvty"); recvty x(U);
        println("rhs recv"); recv x(y);
        println("rhs send"); send x(z.
            println("fwd"); fwd y z
        );
        println("rhs close"); close x
    }
};;
