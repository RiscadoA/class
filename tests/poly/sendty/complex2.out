sendty
recvty
lhs send
let y
rhs recv
rhs send
rhs close
fwd
lhs recv
lhs wait
7
